lib/maths/CBoostedTreeFactory.cc

```diff
@@ -971,8 +979,10 @@ void CBoostedTreeFactory::initializeTrainingProgressMonitoring() {
     //  - The cost of the final train which we count as an extra loop.
 
     std::size_t totalNumberSteps{2};
-    std::size_t lineSearchMaximumNumberTrees{
-        computeMaximumNumberTrees(m_TreeImpl->m_Eta)};
+    double eta{m_TreeImpl->m_EtaOverride != boost::none
+                   ? *m_TreeImpl->m_EtaOverride
+                   : computeEta(frame.numberColumns())};
```

Comment:
There is an order problem: we need to set up progress monitoring before starting `initializeHyperparameters`. However, this needs to anticipate the correct value for `eta` so we monitor progress "correctly". In fact, there is still a problem because `eta` gets set to different values in the hyperparameter optimisation loop, but this is at least better than what we had before. (I want to re-evaluate our strategy for `eta` in a following PR, at which point I'll also fully fix progress monitoring.)